### ISSUE TITLE
Signal color space in image formats using ICC Profiles

### ISSUE THREAD

**Issue by simontWork** [w3c/ColorWeb-CG#9]
I think most are in agreement that magic strings and invaild ICC profiles are not a good way of handling still image formats.

Could we explore:

1. Creating ICCMax profiles (with the help of ICC), or
2. using a floating point linear image format, such as that described in ITU-R BT.2100?

**Comment by palemieux** [comment]
What is the context of this issue?

**Comment by simontWork** [comment]
Still image formats are discussed in the document, but no standards compliant method is outlined.

**Comment by palemieux** [comment]
> but no standards compliant method is outlined.

What method(s) do you have in mind?

* signaling the color space information
* carrying the color space information and the image pixels
* encoding the image pixels

**Comment by digitaltvguy** [comment]
This issue is being discussed in ICC HDR Working Group.  Is there a liason to ICC in W3C?

**Comment by ProgramMax** [comment]
FWIW Chrome already supports ICC profiles in most image formats.
When you say "still image formats" I assume this is as opposed to video? But I think animated images might also belong in this group.

I would like to add a restriction to ICC profile use in file formats: the ICC profile must come before the frame data. Mirroring what I said in #11, web browsers display the image as it is being downloaded. If the image is fully displayed before the browser learns the image should be in a different color space, the user will experience a visual pop.

For animated images like GIF, APNG, and animated WebP, I might need the wisdom of the entertainment folks. Would it be safe to allow only one ICC profile for a whole image? Would a movie have one scene in color space A and another in color space B? Or a TV channel where one show is color space A and the next is color space B?

Once I learn more about that, I can provide some insight into what would happen if the ICC profile comes between frames of an animated image.

Chrome currently does not support ICC profiles in GIFs because all of this is under-specified. There also doesn't seem to be an established "street spec" that everyone agrees upon. I previously contacted the US Library of Congress to recommend they reconsider ICC profile support in GIF as one of its benefits.

**Comment by svgeesus** [comment]
> I think most are in agreement that magic strings and invaild ICC profiles are not a good way of handling still image formats.

Yes. (The context is [Using the ITU BT.2100 PQ EOTF with the PNG Format](https://www.w3.org/TR/png-hdr-pq/) which embeds an invalid ICC profile, basically as a flag).

**Comment by svgeesus** [comment]
> This issue is being discussed in ICC HDR Working Group. 

It is, yes.

> Is there a liaison to ICC in W3C?

Yes, W3C is an ICC Associate Member. I'm the W3C ICC representative, and vice-chair of the ICC HDR WG.

**Comment by svgeesus** [comment]
> For animated images like GIF, APNG, and animated WebP, I might need the wisdom of the entertainment folks. Would it be safe to allow only one ICC profile for a whole image?

In general, only a single profile is supported. I haven't seen any way to apply per-scene profiles in any video or image format.

> Chrome currently does not support ICC profiles in GIFs because all of this is under-specified. 

Yes in general the GIF specification is under-specified. It even states that it must not be used for animation :)

**Comment by svgeesus** [comment]
> I would like to add a restriction to ICC profile use in file formats: the ICC profile must come before the frame data. 

I suspect that would not be Web-compatible (there are a lot of tagged still images out there, and unless the image format and the encoding software agree that the profile comes before the image data, then it could be anywhere).

However, it is certainly a good principle. And it could be enforced like this:

- browsers **must** apply a profile that comes before the image/frame data
- browsers **may** apply a profile that comes after the image/frame data

Together with an update to common image-optimization tools, that would soon result in profiles appearing before the image data.

**Comment by simontWork** [comment]
As @svgeesus states, the initial issue was the use of a fake ICC profile to signal HDR, requiring a special decoder.

One other method for discussion is the CICP method, copying the video signaling from ITU/SM, which may remove the need for complex ICC or a new image format.

**Comment by svgeesus** [comment]
The ICC approach is more flexible, extensible, but ICC Max is currently not well deployed and it is not clear that ICC v.4 will do the job (it may, but that needs to be explored).

The CICP method is currently unspecified and image formats would need to be updated to add it. It is less flexible and not extensible (unless the underlying broadcast standard is updated) but covers most/all use cases. It could tie into existing hardware decoders.

**Comment by ProgramMax** [comment]
> I suspect that would not be Web-compatible (there are a lot of tagged still images out there...
You might be right about this. I seem to recall when Chrome added ICC profile support to images, we did a check to see how many images don't match what we would expect. My memory is that an extremely low percent placed ICC profile after frame data, so we went with it. I also seem to recall GIFs with ICC profiles being exceedingly rare.

I can double check these if we want to consider not having the "may" clause. But I'm not terribly worried about it because the "may" clause might end up going unused anyway.


> the initial issue was the use of a fake ICC profile to signal HDR
I apologize. I misunderstood and accidentally hijacked the conversation.
I believe it would be fine for Chrome to honor other color profiles.
